FancyBatching.md

```diff
@@ -0,0 +1,505 @@
+         Exception exception,
+         CancellationToken cancellationToken) =>
+        {
+            // Remove action
+        };
+    await indexer.UploadDocumentsAsync(products);
+    ```
+
+## Obtaining a SearchIndexingBufferedSender
+
+- `SearchClient.CreateIndexingBufferedSender<T>(SearchIndexingBufferedSenderOptions<T>)`
+    - Instantiates a new `SearchIndexingBufferedSender` tied to the current client.
+    - The sender knows about the type of documents in its index
+        - Could also be used with `SearchDocument` for unknown document types
+    - The only way of obtaining a `SearchIndexingBufferedSender` (i.e., no .ctor)
+    - `virtual` as the entry point for mocking fancy batching scenarios
+
+## Configuring SearchIndexingBufferedSender
+
+- `public class SearchIndexingBufferedSenderOptions<T> { ... }`
+
+    - The generic parameter `T` is the type of documents the
+    `SearchIndexingBufferedSender` will manage.  At one point this tyep had to
+    be generic for the events, but now it's only for the `KeyFieldAccessor`
+    which could be simplified to take `object` if we wanted to remove the type
+    parameter here.
+
+    - `bool AutoFlush { get; set; } = true;`
+
+    - `TimeSpan? AutoFlushInterval { get; set; } = TimeSpan.FromSeconds(60);`
+
+    - `CancellationToken AutoFlushIntervalCancellationToken { get; set; }`
+    Specifically for when the auto flush happens from the interval elapsed timer.
+
+    - `Func<T, string> KeyFieldAccessor { get; set; }`  Used to get the key of
+    a document in case we need to associate errors with their inputs.  We'll use
+    this first if provided.
+
+    - `private int? BatchSize  { get; set; }` This will refer to the length of
+    the JSON payload and not the number of documents.  We'll keep this private
+    for the time being until we can provide reasonable tuning.
+
+    - `private int RetryCount { get; set; } = 3;`  The number of manual retries
+    we'll perform for failed documents.  We default to the same as Azure.Core
+    which is 3 retries and keep it private for now while we think about how to
+    expose the tuning of these parameters.
+
+    - `private ResponseClassifier RetryClassifier { get; set; }`  This is the
+    type we use in Azure.Core to customize Retry behavior.  It will basically
+    retry on 422, 409, 503 status codes and the standard IOException and
+    RequestFailedException with status code 0.
+
+    - TODO: If we end up using PID Controllers for control stabilization, should
+    we expose their gain configuration?
+
+## Using SearchIndexingBufferedSender
+
+- `public class SearchIndexingBufferedSender<T>`
+    - Add Indexing Actions
+        - `void DeleteDocuments(IEnumerable<T>, CancellationToken)`
+        - `Task DeleteDocumentsAsync(IEnumerable<T>, CancellationToken)`
+        - `void IndexDocuments(IndexDocumentsBatch<T>, CancellationToken)`
+        - `Task IndexDocumentsAsync(IndexDocumentsBatch<T>, CancellationToken)`
```

Comment:
Java opted to take a `List<IndexAction<T>>` instead of the `IndexDocumentBatch<T>`

Reply:
As long as you (and everyone else) matches your `SearchClient` APIs, I think that's the right call.